variants/Generic_F401Rx/ldscript.ld

```diff
@@ -0,0 +1,186 @@
+/*
+*****************************************************************************
+**
+
+**  File        : LinkerScript.ld
+**
+**  Abstract    : Linker script for STM32F401RETx Device with
+**                512KByte FLASH, 96KByte RAM
+**
+**                Set heap size, stack size and stack location according
+**                to application requirements.
+**
+**                Set memory bank area and size if external memory is used.
+**
+**  Target      : STMicroelectronics STM32
+**
+**
+**  Distribution: The file is distributed as is, without any warranty
+**                of any kind.
+**
+*****************************************************************************
+** @attention
+**
+** <h2><center>&copy; COPYRIGHT(c) 2014 Ac6</center></h2>
+**
+** Redistribution and use in source and binary forms, with or without modification,
+** are permitted provided that the following conditions are met:
+**   1. Redistributions of source code must retain the above copyright notice,
+**      this list of conditions and the following disclaimer.
+**   2. Redistributions in binary form must reproduce the above copyright notice,
+**      this list of conditions and the following disclaimer in the documentation
+**      and/or other materials provided with the distribution.
+**   3. Neither the name of Ac6 nor the names of its contributors
+**      may be used to endorse or promote products derived from this software
+**      without specific prior written permission.
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+** AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+** IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+** FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+** DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+** CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+** OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+*****************************************************************************
+*/
+
+/* Entry Point */
+ENTRY(Reset_Handler)
+
+/* Highest address of the user mode stack */
+_estack = 0x20018000;    /* end of RAM */
```

Comment:
The linker script have to be updated to use FLASH and RAM size definition.
Example:
https://github.com/stm32duino/Arduino_Core_STM32/blob/2560d7b8090e491c3f853c116077abcc5f0b5e6b/variants/BLACK_F407XX/ldscript.ld#L55-L65

Reply:
I was hoping this pinout was generic enough to be used with F401RB, F401RC, F401RD and F401RE. Is there any way we can make the script "detect" what target is selected in boards.txt? would it be better if I created a generic pinout for each *RB -> *RE variant?

Reply:
Any comments on this? This is my biggest worry. It's also important that it will work in other environments such as PlatformIO too.

Reply:
Well I've already answered.
Use those definitions:
```
LD_MAX_DATA_SIZE
LD_FLASH_OFFSET
LD_MAX_SIZE 
```

Then for all will be properly set.
The linker script is pre-processed and then all values will be set to the linked variant selected based on definition in the boards.txt.
https://github.com/stm32duino/Arduino_Core_STM32/blob/2560d7b8090e491c3f853c116077abcc5f0b5e6b/platform.txt#L41